### ISSUE TITLE
EOFException in cloud endpoints java client

### ISSUE THREAD

**Issue by wonderfly** [google/google-http-java-client#230]
_From [supp...@stacktype.com](https://code.google.com/u/109365404105658033239/) on June 09, 2013 21:35:45_

Version of google-http-java-client (e.g. 1.15.0-rc)? google-api-client-1.14.1-beta.jar
google-api-client-android-1.14.1-beta.jar
google-http-client-1.14.1-beta.jar
google-http-client-android-1.14.1-beta.jar
google-http-client-gson-1.14.1-beta.jar
google-oauth-client-1.14.1-beta.jar Java environment (e.g. Java 6, Android 2.3, App Engine)? Android 4.2.2, With App python engine backend 1.8 using cloud endpoints. Describe the problem. I have an android app that uses a cloud endpoints python backend. I get an EOFException in my android app for some requests. But the same requests, when retried don't give the EOFException.This does not happen for all request, but only for 5-10&#37; of them.

06-10 03:53:45.057: W/System.err(1460):     at android.os.AsyncTask$2.call(AsyncTask.java:287)
06-10 03:53:45.057: W/System.err(1460):     at java.util.concurrent.FutureTask.run(FutureTask.java:234)
06-10 03:53:45.057: W/System.err(1460):     at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:230)
06-10 03:53:45.057: W/System.err(1460):     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1080)
06-10 03:53:45.057: W/System.err(1460):     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:573)
06-10 03:53:45.057: W/System.err(1460):     at java.lang.Thread.run(Thread.java:856)
06-10 03:53:45.107: W/System.err(1460): Caused by: java.io.EOFException
06-10 03:53:45.167: W/System.err(1460):     at libcore.io.Streams.readAsciiLine(Streams.java:203)
06-10 03:53:45.197: W/System.err(1460):     at libcore.net.http.HttpEngine.readResponseHeaders(HttpEngine.java:573)
06-10 03:53:45.288: W/System.err(1460):     at libcore.net.http.HttpEngine.readResponse(HttpEngine.java:821)
06-10 03:53:45.297: W/System.err(1460):     at libcore.net.http.HttpURLConnectionImpl.getResponse(HttpURLConnectionImpl.java:283)
06-10 03:53:45.397: W/System.err(1460):     at libcore.net.http.HttpURLConnectionImpl.getResponseCode(HttpURLConnectionImpl.java:495)
06-10 03:53:45.677: W/System.err(1460):     at libcore.net.http.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:134)
06-10 03:53:46.037: W/System.err(1460):     at com.google.api.client.http.javanet.NetHttpResponse.<init>(NetHttpResponse.java:37)
06-10 03:53:46.057: W/System.err(1460):     at com.google.api.client.http.javanet.NetHttpRequest.execute(NetHttpRequest.java:95)
06-10 03:53:46.057: W/System.err(1460):     at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:980)
06-10 03:53:46.097: W/System.err(1460):     at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:412)
06-10 03:53:46.198: W/System.err(1460):     at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:345)
06-10 03:53:46.198: W/System.err(1460):     at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:463)
06-10 03:53:46.247: W/System.err(1460):     at com.androidapp.android.ServerUtilities.postMessage(ServerUtilities.java:154) How would you expect it to be fixed? I would appreciate if there is a workaround till this issue is fixed.

_Original issue: http://code.google.com/p/google-http-java-client/issues/detail?id=230_

**Comment by bropane** [comment]
Maybe this will help someone but I was having this issue too. I switched to ApacheHttpTransport and got an exception that said my server wasn't responding instead of the EOFException. It was intermittent just like the EOFException so I figured it was the same error causing it but a more concise message. So then I assumed that I was just doing too much work in my server's api method. I changed my backend code to be more efficient and switched back to AndroidHttp.newCompatibleTransport() and haven't had any problems since.
